### PR TITLE
[DX-1221] Annotate exposed functions & types

### PR DESCRIPTION
**Pull request by matthewmuscat**
# Summary
This PR adds typedoc annotations for all publicly exposed functions and types. It also improves consistency with file name casing and other minor upgrades.

# Why the changes
Better DX

# Things worth calling out

### CODE REVIEW

File: src/types/generic.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  registrationContractAddress: string;
+  chainID: number;
+}
+
+/**
+ * The configuration for the ImmutableX client
+ */
+export interface ImmutableXConfiguration {
+  /**
+   * The configuration for L2
```
Comment (Kastet) [inline src/types/generic.ts:52]:
```suggestion
   * The configuration for the API client
```

File: src/types/requests.ts [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
-   * @memberof NftTransferDetails
-   */
-  receiver: string;
-  tokenId: string;
-  tokenAddress: string;
-}
+/**
+ * Parameter required to mint tokens
+ */
+export type UnsignedMintRequest = Omit<MintRequest, 'auth_signature'>;
```
Comment (Kastet) [inline src/types/requests.ts:43]: this type can probably moved to `minting.ts`?

File: src/types/requests.ts [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
 
-// These custom interfaces are used because API.SignableToken is not yet a union type due to OAS 2.0 spec not supporting `OneOf`
+// These custom request interfaces are used because API.SignableToken is not yet a union type due to OAS 2.0 spec not supporting `OneOf`
 // As well as the fact that SignableToken has an extra `data` object which we would like to flatten
 // SignableToken is replaced with AnyToken
 
+/**
+ * Parameter required to create an Order
+ */
 export interface UnsignedOrderRequest {
```
Comment (Kastet) [inline src/types/requests.ts:11]: this type can probably moved to `orders.ts`?

File: src/types/requests.ts (hunk 2 above)
Comment (Kastet) [reply src/types/requests.ts:43]: not blocking

File: src/types/requests.ts (hunk 3 above)
Comment (Kastet) [reply src/types/requests.ts:11]: same applies to all other request types in this file. they can probably belong in their respective workflow files. but not blocking

File: src/types/requests.ts [hunk 4]
```diff
@@ -1,43 +1,43 @@
...
-   * Ethereum address of the receiving user
-   * @type {string}
-   * @memberof NftTransferDetails
-   */
-  receiver: string;
-  tokenId: string;
-  tokenAddress: string;
-}
+/**
+ * Parameter required to mint tokens
```
Comment (Kastet) [inline src/types/requests.ts:41]:
```suggestion
 * Parameters required to mint tokens
```

File: src/types/tokens.ts [hunk 5]
```diff
@@ -1,33 +1,64 @@
...
+   */
   amount: string;
 }
 
+/**
+ * The token details and amount of ERC20 token units
+ */
 export interface ERC20Amount extends ERC20Token {
+  /**
+   * An amount in ERC20 unit
```
Comment (Kastet) [inline src/types/tokens.ts:42]:
```suggestion
   * An amount in units for the given ERC20 token
```

File: src/types/requests.ts (hunk 4 above)
Comment (matthewmuscat) [reply src/types/requests.ts:41]: Hmm, I thought about this. The interface is technically a `parameter`.. that contains `parameters`. I think non-plural makes sense?

File: src/types/tokens.ts (hunk 5 above)
Comment (matthewmuscat) [reply src/types/tokens.ts]: https://github.com/immutable/imx-core-sdk/pull/170/commits/91095159d2be8a9cc717fe479c585fe6ab193eed

File: src/types/generic.ts (hunk 1 above)
Comment (matthewmuscat) [reply src/types/generic.ts]: https://github.com/immutable/imx-core-sdk/pull/170/commits/91095159d2be8a9cc717fe479c585fe6ab193eed

File: src/ImmutableX.ts [hunk 6]
```diff
@@ -89,74 +92,131 @@ export class ImmutableX {
   }
 
   /**
-   * Deposits
+   * Deposit based on a token type
```
Comment (Kastet) [inline src/ImmutableX.ts:95]: Will improve workflow descriptions later to be more useful.

Comment (Kastet) [inline src/ImmutableX.ts:95]:
```suggestion
   * Deposit based on a token type. For unregistered users deposit will be combined with registration in order to register the user first.


```

File: src/ImmutableX.ts [hunk 7]
```diff
@@ -89,74 +92,131 @@ export class ImmutableX {
...
+   * @param ethSigner - the L1 signer
+   * @param deposit - the token type amount in its corresponding unit
+   * @returns the transaction
```
Comment (Kastet) [inline src/ImmutableX.ts:98]:
```suggestion
   * @returns the response of the resulting transaction
```

File: src/ImmutableX.ts [hunk 8]
```diff
@@ -89,74 +92,131 @@ export class ImmutableX {
...
   public deposit(ethSigner: EthSigner, deposit: TokenAmount) {
     return this.workflows.deposit(ethSigner, deposit);
   }
 
+  /**
+   * Get details of a Deposit with the given ID
+   * @param requestParameters - the request object containing the parameters to be provided in the API request
+   * @returns the requested Deposit
+   */
   public getDeposit(requestParameters: DepositsApiGetDepositRequest) {
```
Comment (Kastet) [inline src/ImmutableX.ts:109]: @matthewmuscat `requestParameters` -> `request`?

Comment (matthewmuscat) [reply src/ImmutableX.ts:109]: I believe the idea is to use `request` for all POST actions (creates/etc) and `requestParameters` for everything else. Although I don't know why we can't just make it `request` for everything?

File: src/ImmutableX.ts [hunk 9]
```diff
@@ -89,74 +92,131 @@ export class ImmutableX {
...
+   * @returns the requested list of Deposits
+   */
   public listDeposits(requestParameters?: DepositsApiListDepositsRequest) {
     return this.depositsApi.listDeposits(requestParameters);
   }
 
   /**
-   * User Registration
+   * Register a User to Immutable X if they are not already
+   * @param walletConnection - the pair of Signers for the wallet
```
Comment (Kastet) [inline src/ImmutableX.ts:124]:
```suggestion
   * @param walletConnection - the pair of Signers
```

Comment (Kastet) [reply src/ImmutableX.ts:124]: (if applied, update all instances)

File: src/ImmutableX.ts [hunk 10]
```diff
@@ -89,74 +92,131 @@ export class ImmutableX {
...
+   * Checks if a User is registered on on-chain
+   * @param walletConnection - the pair of Signers for the wallet
+   * @returns true if the User is registered, false otherwise
+   */
   public isRegisteredOnchain(walletConnection: WalletConnection) {
     return this.workflows.isRegisteredOnchain(walletConnection);
   }
 
+  /**
+   * Get stark keys for a registered User
```
Comment (Kastet) [inline src/ImmutableX.ts:141]:
```suggestion
   * Get Stark keys for a registered User
```

File: src/ImmutableX.ts [hunk 11]
```diff
@@ -214,10 +294,25 @@ export class ImmutableX {
...
+   * @param ethSigner - the L1 signer
+   * @param id - the Project ID
+   * @returns the requested Project
+   */
   public async getProject(ethSigner: EthSigner, id: string) {
     return this.workflows.getProject(ethSigner, id);
   }
 
+  /**
+   * Get Projects
```
Comment (Kastet) [inline src/ImmutableX.ts:308]:
```suggestion
   * Get projects owned by the given user
```

File: src/ImmutableX.ts (hunk 8 above)
Comment (Kastet) [reply src/ImmutableX.ts:109]: No reason. We can call it `request` in the public interface.

**Review by Kastet — APPROVED** [review]
Overall LGTM but mindful that another round of improving workflow descriptions might be required. Left a few suggestions.

File: src/ImmutableX.ts (hunk 11 above)
Comment (matthewmuscat) [reply src/ImmutableX.ts]: https://github.com/immutable/imx-core-sdk/pull/170/commits/c6328988252e1f652abc2ef3afed662a0f907ba4

File: src/ImmutableX.ts (hunk 10 above)
Comment (matthewmuscat) [reply src/ImmutableX.ts]: https://github.com/immutable/imx-core-sdk/pull/170/commits/c6328988252e1f652abc2ef3afed662a0f907ba4

File: src/ImmutableX.ts (hunk 9 above)
Comment (matthewmuscat) [reply src/ImmutableX.ts]: https://github.com/immutable/imx-core-sdk/pull/170/commits/c6328988252e1f652abc2ef3afed662a0f907ba4

File: src/ImmutableX.ts (hunk 8 above)
Comment (matthewmuscat) [reply src/ImmutableX.ts]: https://github.com/immutable/imx-core-sdk/pull/170/commits/c6328988252e1f652abc2ef3afed662a0f907ba4

File: src/ImmutableX.ts [hunk 12]
```diff
@@ -332,34 +493,60 @@ export class ImmutableX {
...
+   */
   public transfer(
     walletConnection: WalletConnection,
     request: UnsignedTransferRequest,
   ) {
     return this.workflows.transfer(walletConnection, request);
   }
 
+  /**
+   * Create a batch of transfer requests
```
Comment (zatlite) [inline src/ImmutableX.ts:545]: probably specify this is only for NFTs

**Pull request merged**